src/bepasty/utils/upload.py

```diff
@@ -82,7 +92,10 @@ def meta_new(cls, item, input_size, input_filename, input_type,
             input_filename, storage_name, input_type, input_type_hint
         )
         item.meta[SIZE] = cls.filter_size(input_size)
-        item.meta[TYPE] = cls.filter_type(input_type, input_type_hint, input_filename)
+        ct, hint = cls.filter_type(input_type, input_type_hint, input_filename)
+        item.meta[TYPE] = ct
+        if hint and mag and current_app.config.get('USE_PYTHON_MAGIC'):
```

Comment:
code works due to None being falsy, but slightly cleaner is: `... .get('USE_PYTHON_MAGIC', False):`

Reply:
guess you could also **not** do this conditional here and just write hint to the metadata, no matter whether True or False.

then, when the item is completed, you remove the hint again anyway and you can do the conditional there only.

Reply:
hm, im not sure i got comment correctly though. if hint is False, we dont need to add TYPE_HINT field...

Reply:
yes. but as you remove it later anyway, it would not do harm and simplify the code.

Reply:
right, however this is on-disk data. so it would be better to keep as possible as smaller?

Reply:
it's only temporarily larger and only a few bytes, so it doesn't matter.

Reply:
so do it however you prefer, i am ok with both ways. tell me when finished so i can merge this.

Reply:
ok, i changed to store unconditionally